dvc/repo/experiments/executor/base.py

```diff
@@ -248,6 +250,30 @@ def on_diverged_ref(orig_ref: str, new_rev: str):
             )
         return refs
 
+    @classmethod
+    def _validate_remotes(cls, dvc: "Repo", git_remote: Optional[str]):
+        from dulwich.client import get_transport_and_path
+        from dulwich.porcelain import get_remote_repo
+
+        from dvc.scm.base import SCMError
+
+        if git_remote == dvc.root_dir:
+            logger.warning(
+                f"'{git_remote}' points to the current Git repo, experiment "
+                "Git refs will not be pushed. But DVC cache and run cache "
+                "will automatically be pushed to the default DVC remote "
+                "(if any) on each experiment commit."
+            )
+        try:
+            _, location = get_remote_repo(dvc.scm.dulwich.repo, git_remote)
```

Comment:
One problem here. This method not only uses low-level Dulwich API but also can not raise proper Exception with a Git remote value `true`

Reply:
Right, this should not go here. It needs to be added as a new `dvc.scm.git.backend` call (with the non-dulwich implementations set to raise `NotImplementedError`)

Reply:
> Right, this should not go here. It needs to be added as a new `dvc.scm.git.backend` call (with the non-dulwich implementations set to raise `NotImplementedError`)

Another problem is that these lines will not raise any exception even if I set the git_remote to a none URL format (guess, this would be considered to some references of a URL). The exception only raises when it began to read out the data from the remote repo.

Reply:
It's because dulwich just falls back to assuming `location` is supposed to be a local path (and assumes there will be a valid git repo at `file:///location`.
https://github.com/dulwich/dulwich/blob/952573478aa2b0833b8a51b7e741e9e13fc14060/dulwich/client.py#L2210

If dulwich returns a local client you can just check that the local path actually exists with:

```py
from dulwich.client import LocalGitClient
_, location = get_remote_repo(dvc.scm.dulwich.repo, git_remote)
client, path = get_transport_and_path(location)
if isinstance(client, LocalGitClient) and not os.path.exists(os.path.join("", path)):
    # this is an invalid local client/path
    raise ...
```